doc/designs/quic-design/quic-overview.md

```diff
@@ -0,0 +1,128 @@
+QUIC Design Overview
+====================
+
+The QUIC implementation in OpenSSL is roughly described by the following
+picture.
+
+![alt_text](images/quic-overview.svg "QUIC Implementation Building Blocks")
+
+SSL API
+-------
+
+The application facing public API of the OpenSSL library.
+
+Stream Send and Read Buffers
+----------------------------
+
+Buffers for stream data to be sent or received from the peer over the
+QUIC protocol. These are necessary to support existing semantics of the
+SSL_read and SSL_write functions.
+
+They will be bypassed with a single-copy API for read and write (_not
+for MVP_).
+
+Connection State Machine
+------------------------
+
+A state machine handling the state for a QUIC connection.
+
+Connection ID Cache
+-------------------
+
+A table matching Connection IDs with Connection objects represented
+via SSL objects.
+
+_In MVP there is basically 1-to-1 matching of Connection IDs to Connection
```

Comment:
It's not quite 1 to 1.  Each end of the connection has a different ID so 2 to 1 but in different ID spaces.  Sending packets uses a different ID to receiving them.  I guess we won't need to loop up by remote ID for MVP though.